src/lines-component.coffee

```diff
@@ -19,11 +20,7 @@ class LinesComponent
   placeholderTextDiv: null
 
   constructor: ({@presenter, @hostElement, @useShadowDOM, visible}) ->
```

Comment:
Like I said in chat, I'm skeptical about 1. We're allocating plenty of objects in the typical render frame. One per tile doesn't seem to represent a significant increase.

I *would* like to see you try out 2, because in the timeline it definitely seems like there's overhead for the discrete parse calls. You could try out the wrapper div approach and just generate all the DOM nodes in masse, then populate the tiles. Not sure it will be a win by the time we do all that, but worth a look.

Not sure about 3. Probably easy to experiment with and see how much it helps?